completion_aggregator/tasks/aggregation_tasks.py

```diff
@@ -201,16 +206,27 @@ def update_for_aggregator(self, block, affected_aggregators, force):
             if modified is not None:
                 last_modified = max(last_modified, modified)
         if self._aggregator_needs_update(block, last_modified, force):
-            obj, _ = Aggregator.objects.submit_completion(
-                user=self.user,
-                course_key=self.course_key,
-                block_key=block,
-                aggregation_name=block.block_type,
-                earned=total_earned,
-                possible=total_possible,
-                last_modified=last_modified,
-            )
-            self.aggregators[obj.block_key] = obj
+
+            Aggregator.objects.validate(self.user, self.course_key, block)
+            aggregator_identifier_key = '$'.join(map(str, [self.user.id, self.course_key, block, block.block_type]))
+            if aggregator_identifier_key not in self.aggregator_map:
```

Comment:
I don't see why this needs to be so complicated. You can just do `if block not in self.aggregators:`

Reply:
I thought I should use what makes an `Aggregator` object unique. Updating this to use `block` only.